controllers/configurationpolicy_controller.go

```diff
@@ -2991,6 +2992,17 @@ func (r *ConfigurationPolicyReconciler) updatePolicyStatus(
 			log.Info(fmt.Sprintf("Failed to refresh policy; using previously fetched version: %s", err))
 		} else {
 			policy.Status = updatedStatus
+
+			// If the policy was recreated after it was retrieved from the cache, there could be no compliance
+			// set on the parent policy. Updating the status here may prevent a compliance event from being generated on
+			// future evaluations so skipping the status update is very important. If sendEvent was true, there is still
+			// value to send the compliance event to the parent policy so that this history is not lost before the
+			// policy was recreated.
```

Comment:
Maybe: "If the UID has changed, then the policy has been deleted and created again. Do not update the status, because it was calculated based on a previous version. If sendEvent is true, that event might be useful and it can be emitted. By leaving the status blank, the policy will be reevaluated and send a new event.